### ISSUE TITLE
Need to reinstall planemo every time after using it

### ISSUE THREAD

**Issue by bebatut** [galaxyproject/planemo#1078]
Here is what I did to setup planemo
- I created a conda environment with only Python 3.6 (`conda create -n planemo python=3.6`) 
- I installed Planeno in it with pip (`pip install planemo`)

When I run Planeno for test (`planemo test --galaxy_python_version 3.6 --galaxy_branch release_20.05 .`), it works fine the 1st time (installing Galaxy, running the Galaxy and the tests). But if I run it a 2nd time, I got this error:

```
Traceback (most recent call last):
  File "~/miniconda3/envs/planemo/bin/planemo", line 5, in <module>
    from planemo.cli import planemo
  File "~/miniconda3/envs/planemo/lib/python3.6/site-packages/planemo/cli.py", line 11, in <module>
    from planemo.galaxy import profiles
  File "~/miniconda3/envs/planemo/lib/python3.6/site-packages/planemo/galaxy/__init__.py", line 5, in <module>
    from .config import galaxy_config
  File "~/miniconda3/envs/planemo/lib/python3.6/site-packages/planemo/galaxy/config.py", line 52, in <module>
    from .workflows import (
  File "~/miniconda3/envs/planemo/lib/python3.6/site-packages/planemo/galaxy/workflows.py", line 12, in <module>
    from gxformat2.normalize import inputs_normalized, outputs_normalized
ImportError: cannot import name 'outputs_normalized'
```

If I reinstall Planeno with pip, it works again. And then the next time, same error. Need to reinstall Planeno.

It seems that a different version of `gxformat2` is reinstalled:

```
Installing collected packages: bioblend, gxformat2
  Attempting uninstall: bioblend
    Found existing installation: bioblend 0.13.0
    Uninstalling bioblend-0.13.0:
      Successfully uninstalled bioblend-0.13.0
  Attempting uninstall: gxformat2
    Found existing installation: gxformat2 0.11.3
    Uninstalling gxformat2-0.11.3:
      Successfully uninstalled gxformat2-0.11.3
Successfully installed bioblend-0.14.0 gxformat2-0.15.
```

Is it an expected behavior? Are Galaxy and Planemo using different versions of `gxformat2` and `gxformat2` is overwriting when Galaxy is installed during planemo test?

**Comment by mvdbeek** [comment]
Can you post the logs from when you run the first test ?

**Comment by bebatut** [comment]
I just saw an error in the logs:

```
Installing collected packages: bioblend, gxformat2
  Attempting uninstall: bioblend
    Found existing installation: bioblend 0.14.0
    Uninstalling bioblend-0.14.0:
      Successfully uninstalled bioblend-0.14.0
  Attempting uninstall: gxformat2
    Found existing installation: gxformat2 0.15.0
    Uninstalling gxformat2-0.15.0:
      Successfully uninstalled gxformat2-0.15.0
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

planemo 0.72.0 requires bioblend>=0.14.0, but you'll have bioblend 0.13.0 which is incompatible.
planemo 0.72.0 requires gxformat2>=0.12.0, but you'll have gxformat2 0.11.3 which is incompatible.
Successfully installed bioblend-0.13.0 gxformat2-0.11.3
Looking in indexes: https://wheels.galaxyproject.org/simple, https://pypi.python.org/simple
```

It is line 275 in full logs: https://gist.github.com/bebatut/1d8f0bc8de8001a3829cd27086ca7501

**Comment by mvdbeek** [comment]
The planemo virtualenv and the Galaxy virtualenv are supposed to be separate.
Can you run `type planemo` before running the next test ?
It ... almost looks like planemo is running from the virtualenv it created to install Galaxy's dependencies in?

**Comment by fubar2** [comment]
Interesting re #1097 ?

tl;dr seems to be some leakage into the current python venv from planemo testing when not using conda

If a test run uses (for example) an older Galaxy with a previous set of requirements, those will be downgraded permanently in the host venv - not what might be expected. In particular, if an older version of gxformat2 is installed then any future attempt at using the planemo in that venv will throw an error about the old version like the one reported above:

`ImportError: cannot import name 'outputs_normalized'`

Here's a test script to show what I'm seeing. 
```
# planemo venv leakage demo
rm -rf planemoenv
rm -rf *.req
rm -rf *.diff
python3 -m venv planemovenv
. planemovenv/bin/activate
pip3 install planemo==0.72.0 bioblend==0.14.0 gxformat2==0.15.0
pip3 list > pre.req
planemo
# test with current dev
planemo test --engine galaxy --galaxy_root ~/galaxy-dev /home/ross/galaxy/tools/tool_makers/sedtest/sedtest.xml 
pip3 list > post.req
diff pre.req post.req > after_first_test.diff
planemo
# this still works in our planemovenv
# next test  leads to leaked downgrade in the planemoenv
planemo test --engine galaxy --galaxy_root ~/galaxy-release_20.09 /home/ross/galaxy/tools/tool_makers/sedtest/sedtest.xml 
pip3 list > post2.req
diff post.req post2.req > after_second_test.diff
echo "#### calling planemo will now fail because gxformat2 got downgraded - but planemo has broken the fresh planemoenv not the ~/.planemo/gx_venv_3 one?"
planemo

```


For the record, I've been running around in circles for weeks trying to figure out why things have been breaking :(

**Comment by mvdbeek** [comment]
If you can test with 0.74.3 that'd be great. It might be fixed by https://github.com/galaxyproject/planemo/pull/1145

**Comment by fubar2** [comment]
Will take a look. For bonus points, if planemo is run as a tool it tries to call run_test.sh with --dev-wheels which fails since it's not defined as a parameter. Running on the command line is fine but as a tool, things are a little different it seems. I'll put a simple example together and file it as an issue. It is really head scratchingly odd.

**Comment by pinin4fjords** [comment]
As per ^^^, we appear to be suffering from this still with 0.74.3, in our CI (which I think we mostly borrowed from you guys).

**Comment by mvdbeek** [comment]
You might need to invalidate the cache, by changing changing the galaxy release or the python version, or updating to match https://github.com/galaxyproject/tools-iuc/blob/master/.github/workflows/pr.yaml

**Comment by pinin4fjords** [comment]
> You might need to invalidate the cache, by changing changing the galaxy release or the python version, or updating to match https://github.com/galaxyproject/tools-iuc/blob/master/.github/workflows/pr.yaml

Did try invalidating the cache without any joy, but yes, updating the actions did it, thanks.
